### PR TITLE
Add fromListWith and merge to DA.TextMap and DA.Next.Map

### PR DESCRIPTION
**Pull request by hurryabit**
Also, resurrect the tests we had for both functions when we still tested
`DA.Map`.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/898)

### CODE REVIEW

**Comment by hurryabit** [comment]
Fixes #881.

File: daml-foundations/daml-ghc/daml-stdlib-src/DA/Next/Map.daml [hunk 1]
```diff
@@ -121,6 +132,16 @@ insert k v (Map t) = Map $ TextMap.insert (keyToText k) v t
 union : MapKey k => Map k v -> Map k v -> Map k v
 union (Map t1) (Map t2) = Map $ TextMap.union t1 t2
 
+merge
```
Comment (cocreature) [inline daml-foundations/daml-ghc/daml-stdlib-src/DA/Next/Map.daml]: Some documentation for this function would be nice (I know what it does but our users might not)

File: daml-foundations/daml-ghc/daml-stdlib-src/DA/TextMap.daml [hunk 2]
```diff
@@ -81,6 +98,36 @@ insert = primitive @"BEMapInsert"
 union : TextMap a -> TextMap a -> TextMap a
 union m1 m2 = foldl (\acc (k, v) -> insert k v acc) m2 (toList m1)
 
+-- | Merge two maps.
```
Comment (cocreature) [inline daml-foundations/daml-ghc/daml-stdlib-src/DA/TextMap.daml]: That comment doesn’t really tell users anything more than the function name.

File: daml-foundations/daml-ghc/daml-stdlib-src/DA/Next/Map.daml (hunk 1 above)
Comment (hurryabit) [reply daml-foundations/daml-ghc/daml-stdlib-src/DA/Next/Map.daml]: Oops, that got lost during copy'n'paste.

**Comment by hurryabit** [comment]
Improved documentation is coming in a separate PR.

**Pull request merged**